bitpacker/src/blocked_bitpacker.rs

```diff
@@ -0,0 +1,146 @@
+use crate::BitUnpacker;
+
+use super::{bitpacker::BitPacker, compute_num_bits};
+
+const BLOCK_SIZE: usize = 128;
+
+/// BlockedBitpacker compresses data in blocks of
+/// 128 elements, while keeping an index on it
+///
+#[derive(Debug, Clone)]
+pub struct BlockedBitpacker {
+    // bitpacked blocks
+    compressed_blocks: Vec<u8>,
+    // uncompressed data, collected until BLOCK_SIZE
+    cache: Vec<u64>,
+    offset_and_bits: Vec<BlockedBitpackerEntryMetaData>,
+}
+
+#[derive(Debug, Clone, Default)]
+struct BlockedBitpackerEntryMetaData {
+    encoded: u64,
```

Comment:
i assume this is to avoid padding and avoid using repr(packed) ?

Reply:
yes I should add some comments, we will never need the full 8byte for offset and repr(packed) would also be slower because of alignment